src/Aspire.Dashboard/Utils/FormatHelpers.cs

```diff
@@ -1,23 +1,29 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Concurrent;
 using System.Globalization;
 using System.Text.RegularExpressions;
 
 namespace Aspire.Dashboard.Utils;
 
 internal static partial class FormatHelpers
 {
-    static string GetLongTimePatternWithMilliseconds()
+    private sealed record MillisecondFormatStrings(string LongTimePattern, string ShortDateLongTimePattern);
+
+    // Use culture name as key. Culture name isn't case sensitive.
+    private static readonly ConcurrentDictionary<string, MillisecondFormatStrings> s_formatStrings = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
Could add `StringComparers.CultureNames`. Consider also naming the field something like `s_formatStringByCultureName` to make the comment obsolete.

Reply:
I don't think it's worth it in this case.